### ISSUE TITLE
search.onSelect would not return selected item if there are items having same title with it

### ISSUE THREAD

**Issue by pokehanai** [Semantic-Org/Semantic-UI#2161]
In my use case there are items having same title and different descriptions.
Then, as the current implementation is like following, onSelect returns only the first item among the same title items.

https://github.com/Semantic-Org/Semantic-UI/blob/master/src/definitions/modules/search.js#L141
```js
            click: function(event) {
              module.debug('Search result selected');
              var
                $result = $(this),
                $title  = $result.find(selector.title).eq(0),  // issue
                $link   = $result.find('a[href]').eq(0),  // issue
                href    = $link.attr('href')   || false,
```

I think to have same title items is common use case, e.g. 
- title: person name, description: address 
- title: song title, description: album

**Comment by pokehanai** [comment]
duplicate: #2135

**Comment by cspeer** [comment]
Err, how is #2135 a duplicate of the issue you are describing?
I'm experiencing the same problem as described above and highlighting search matches in the results doesn't have anything to do with this :D

http://jsfiddle.net/efp8z6Ln/141/ @jlukic  

Thanks a lot!

**Comment by jlukic** [comment]
I think he meant #2162

It's an issue with the design of search using `title` as a unique ID for matching elements.

**Comment by cspeer** [comment]
Oh damnit, I did not find that when I searched.
Anyway, good to know you are taking care of it :) Thanks :+1:

**Comment by jlukic** [comment]
Just closed out #2162, results now inject `id` and pull result from metadata.
